data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/InjectionSetpointAdderImpl.java

```diff
@@ -46,13 +48,19 @@ public InjectionSetpointAdder withSetpoint(double setpoint) {
         return this;
     }
 
+    @Override
+    public InjectionSetpointAdder withUnit(Unit unit) {
+        this.unit = unit;
+        return this;
+    }
+
     @Override
     public NetworkActionAdder add() {
         assertAttributeNotNull(networkElementId, "InjectionSetPoint", "network element", "withNetworkElement()");
         assertAttributeNotNull(setpoint, "InjectionSetPoint", "setpoint", "withSetPoint()");
```

Comment:
check that unit is not null. you'll have to update all existing adders called by adding "withUnit(MEGAWATT)"

Reply:
Done !